From f4976aae4e9281d6f40c116d099d006a8dd17434 Mon Sep 17 00:00:00 2001
From: Christoph Junghans <junghans@votca.org>
Date: Thu, 9 Jan 2020 17:37:09 -0700
Subject: [PATCH] gmxManageMPI.cmake: convert MPI FLAGS to lists

Convert MPI FLAGS to lists as otherwise space will get quoted in
GMX_FLAGS down the road.

Change-Id: Ieee4b68d254261f079d954443322faf040e8b6e9
---
 cmake/gmxManageMPI.cmake        | 2 ++
 cmake/gmxTestMPI_IN_PLACE.cmake | 2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/cmake/gmxManageMPI.cmake b/cmake/gmxManageMPI.cmake
index 0a614a430c..795934c6f3 100644
--- a/cmake/gmxManageMPI.cmake
+++ b/cmake/gmxManageMPI.cmake
@@ -51,7 +51,9 @@ if(GMX_MPI)
       find_package(MPI)
       if(MPI_C_FOUND)
           set(MPI_COMPILE_FLAGS ${MPI_C_COMPILE_FLAGS})
+          separate_arguments(MPI_COMPILE_FLAGS)
           set(MPI_LINKER_FLAGS ${MPI_C_LINK_FLAGS})
+          separate_arguments(MPI_C_LINK_FLAGS)
           include_directories(SYSTEM ${MPI_C_INCLUDE_PATH})
           list(APPEND GMX_COMMON_LIBRARIES ${MPI_C_LIBRARIES})
       endif()
diff --git a/cmake/gmxTestMPI_IN_PLACE.cmake b/cmake/gmxTestMPI_IN_PLACE.cmake
index 1d53dae33b..77f4197bc4 100644
--- a/cmake/gmxTestMPI_IN_PLACE.cmake
+++ b/cmake/gmxTestMPI_IN_PLACE.cmake
@@ -44,7 +44,7 @@ MACRO(GMX_TEST_MPI_IN_PLACE VARIABLE)
   if(NOT DEFINED MPI_IN_PLACE_COMPILE_OK)
     MESSAGE(STATUS "Checking for MPI_IN_PLACE")
 
-    set(CMAKE_REQUIRED_FLAGS ${MPI_COMPILE_FLAGS})
+    list(JOIN MPI_COMPILE_FLAGS " " CMAKE_REQUIRED_FLAGS)
     set(CMAKE_REQUIRED_INCLUDES ${MPI_INCLUDE_PATH})
     set(CMAKE_REQUIRED_LIBRARIES ${MPI_LIBRARIES})
     check_cxx_source_compiles(
-- 
2.24.1

